### DISCUSSION TITLE
gobot alpha 35 - PocketBase via npm

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#4718] [Show and tell]
I posted previously #4497 and #4459 about this. Now it's almost stable.

If anyone is interested in adding PocketBase as a npm package dependency, now you can do it.

https://www.npmjs.com/package/gobot

Try this:

```bash
npx gobot pocketbase --help

# Or a specific version or semver
npx gobot --g-version='0.19.*' pocketbase --help
```

There you go.. it can't get easier than that. `gobot` will automatically select the correct binary for your machine, download it, unpack it, and even manage version updates for you.

Let me know if it works :)

You can also use it as a dependency:

```bash
npm i gobot gobot-pocketbase
```

```ts
const bot = gobot(`pocketbase` ) // will automatically use the gobot-pocketbase version
const process = await bot.run([`serve`])
```

Let me know what you think! This came out of the [PocketHost](https://github.com/pockethost/pockethost) project because PocketHost needs to manage every available version of PocketBase.

For those interested in exploring more, `gobot` is capable of installing just about any binary release on github. Examples:

```bash
npx gobot rclone --help
npx gobot caddy --help
npx gobot pulumi --help
```

And if official support isn't built in, you can use `npx gobot <user>/<repo>` and it will "probably" figure out what to do :)

**Comment by moussa32** [comment]
Great job
I am wondering how it works with migrations and changes happened between versions in pocketbase?

**Reply by benallfree** [reply]
Gobot just downloads binaries, you are responsible for doing the same things as if you had downloaded it yourself. Does that make sense?
